src/main/java/uk/gov/hmcts/reform/sscs/model/hmc/reference/EntityRoleCode.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.sscs.model.hmc.reference;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
+public enum EntityRoleCode {
+    APPELLANT("APEL", Constants.APPLICANT, "Appellant", "", null),
+    APPOINTEE("APIN", "Appointee", "Appointee", "", null),
+    JOINT_PARTY("JOPA", Constants.APPLICANT, "Joint Party", "", null),
+    OTHER_PARTY("OTPA", "Respondent", "Other Party", "", null),
+    RESPONDENT("RESP", "Respondent", "Respondent", "", null),
+    WELFARE_REPRESENTATIVE("WERP", Constants.REPRESENTATIVE, "Welfare Representative", "", PartyRelationshipType.SOLICITOR),
+    LEGAL_REPRESENTATIVE("LGRP", Constants.REPRESENTATIVE, "Legal Representative", "", PartyRelationshipType.SOLICITOR),
+    BARRISTER("BARR", Constants.REPRESENTATIVE, "Barrister", "", PartyRelationshipType.SOLICITOR),
+    INTERPRETER("INTP", "Interpreter", "Interpreter", "", PartyRelationshipType.INTERPRETER),
+    REPRESENTATIVE("RPTT", Constants.REPRESENTATIVE, "Barrister", "", PartyRelationshipType.SOLICITOR),
+    SUPPORT("SUPP", "Support", "Support", "", null),
+    APPLICANT("APPL", Constants.APPLICANT, Constants.APPLICANT, "", null);
+
+    private final String hmcReference;
```

Comment:
Can you add the annotation @JsonValue for hmcReference

Reply:
added